Cahps measures

saquino0827 opened the pull request:
QPPSF-4813 Add CAHPS measures for 2019 to repo.

#### Motivation for change
Cahps measure are being added for the submission and feedback team to begin testing tickets related to cahps measures.

#### What is being changed
- Build measures 2019 script changed to support adding cahps measures. 
- add a 2019 version add cahps measures script
- Cahps measures have been added into `measure-data.json`

#### Release checklist:
Tasks that must be done prior to merging this PR, including testing.

* [X] [Package version updated](https://github.com/CMSgov/qpp-measures-data/blob/master/CONTRIBUTING.md#versioning-publishing-and-creating-new-releases)
* [X] Documentation updated
* [X] Unit tests added/passing
* [X] Verified working locally

##### Associated JIRA tickets:
* https://jira.cms.gov/browse/QPPSF-4813

KyleApfel commented on util/measures/2019/enriched-measures-data-quality.json:
```diff
@@ -2488,7 +2488,7 @@
   },
   {
-    "title": "Colonoscopy Interval for Patients with a History of Adenomatous Polyps \r- Avoidance of Inappropriate Use",
+    "title": "Colonoscopy Interval for Patients with a History of Adenomatous Polyps\\r\\n \r- Avoidance of Inappropriate Use",
```
@lamroger Same issue from the csv adding in `\\r\\n` here. Is this something we need to fix?

harrisj replied:
Seems like somethign might need to fix. Those single \r's also look wrong...

saquino0827 replied:
@KyleApfel @harrisj It looks like it was an issue with `util/measures/2019/quality-measures.csv`. After removing the additional `\r\n` from this line in that file, it reverted back to only having `\r-` only. I am hesitant on changing this as this is how it is for the 2018 version of the measure data.

harrisj commented on measures/2019/measures-data.json:
```diff
@@ -2488,7 +2488,7 @@
   },
   {
-    "title": "Colonoscopy Interval for Patients with a History of Adenomatous Polyps \r- Avoidance of Inappropriate Use",
+    "title": "Colonoscopy Interval for Patients with a History of Adenomatous Polyps\\r\\n \r- Avoidance of Inappropriate Use",
```
Looks like we still have some sort of carriage return (unescaped) in here. Is that intentional?

harrisj commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "qpp-measures-data",
-  "version": "1.22.1",
+  "version": "1.22.3",
```
What happened to 1.22.2?

saquino0827 replied:
I have another PR open. Whichever gets merged first will be incremented correctly. I  only incremented this to `1.22.3` because I already incremented the other PR as `1.22.2`

saquino0827 replied:
I have addressed this and followed the process.

shebryant commented:
I left some small comments, and I have a couple of questions.

It’s not apparent in the code so wondering if, after the changes were implemented,  `npm run build:measures 2019` was run? That is usually the case for measures updates, so that updates to the json aren’t done manually, and validation of the updated measures occurs.

Also curious about `quality-measures.csv`. It shows up in modified files, but do the CAHPS changes require anything in the csv for quality measures to change?

KyleApfel commented:
@shebryant That csv was included because it's needed for the shell script to run successfully to generate the CAHPS measures. For 2019, it didn't exist yet.

shebryant commented:
@KyleApfel Oh I understand that CAHPS csv didn't yet exist and needed to be included, but I was asking about `quality-measures.csv` specifically. That file already exists in master doesn't it? I updated it, and `quality-strata.csv`, 11 days ago.

saquino0827 commented:
@shebryant `npm run build:measures 2019` was run but I made a mistake and excluded adding the additional information from cahps measures. I have corrected this mistake and now the cahps measure changes are there.  The changes made to `quality-measures.csv` were because of a previous comment about additional escaped characters being placed where they shouldn't. 

I've addressed the small comments as well. I reverted the manual change to 1.22.3 and used `npm version patch` to go up to 1.22.2.  Thanks for you comments!

saquino0827 commented on measures/2019/measures-schema.yaml:
```diff
@@ -301,6 +301,7 @@ definitions:
       - geriatrics
       - urgentCare
       - skilledNursingFacility
+      - generalPracticeFamilyMedicine
```
For clarification, This type was added as cahps measures use `generalPracticeFamilyMedicine` and the schema was being invalidated because of this. Please let me know if this change is incorrect.

shebryant replied:
Hi, Sam. If there were no errors generated when you ran the build script, this change looks fine to me.

saquino0827 replied:
Okay! I ran `npm test` and did a npm build for 2019 measures and they passed. I'll go ahead and merge.

saquino0827 commented on scripts/measures/2019/add-cahps-measures.js:
```diff
@@ -0,0 +1,108 @@
+const cahpsMeasuresCsv = process.argv[2];
+const outputFilepath = process.argv[3];
+
+importCahpsMeasures(cahpsMeasuresCsv, outputFilepath);
```
I agree. The file has been renamed to the suggested name.

saquino0827 commented on scripts/measures/build-2019-measures:
```diff
@@ -11,6 +11,9 @@ pi_csv='../../../util/measures/'$currentPerformanceYear'/pi-measures.csv'
 ia_csv='../../../util/measures/'$currentPerformanceYear'/ia-measures.csv'
 
+cahps_csv='../../../util/measures/'$currentPerformanceYear'/cahps-measures.csv'
+cahps_measures='../../../staging/'$currentPerformanceYear'/cahps-measures.json'
```
This has been changed to the suggested naming convention.

shebryant commented:
Thanks so much for making these updates.
